test/integration/node-specific/client_close.test.ts

```diff
@@ -0,0 +1,372 @@
+            });
+          });
+        });
+      });
+    });
+
+    describe('SrvPoller', () => {
+      describe('Node.js resource: Timer', () => {
+        describe('after SRVPoller is created', () => {
+          it('timers are cleaned up by client.close()', async () => {});
+        });
+      });
+    });
+  });
+
+  describe('ClientSession (Implicit)', () => {
+    describe('Server resource: LSID/ServerSession', () => {
+      describe('after a clientSession is implicitly created and used', () => {
+        it('the server-side ServerSession is cleaned up by client.close()', async function () {});
+      });
+    });
+
+    describe('Server resource: Transactions', () => {
+      describe('after a clientSession is implicitly created and used', () => {
+        it('the server-side transaction is cleaned up by client.close()', async function () {});
+      });
+    });
+  });
+
+  describe('ClientSession (Explicit)', () => {
+    describe('Server resource: LSID/ServerSession', () => {
+      describe('after a clientSession is created and used', () => {
+        it('the server-side ServerSession is cleaned up by client.close()', async function () {});
+      });
+    });
+
+    describe('Server resource: Transactions', () => {
+      describe('after a clientSession is created and used', () => {
+        it('the server-side transaction is cleaned up by client.close()', async function () {});
+      });
+    });
+  });
+
+  describe('AutoEncrypter', () => {
+    const metadata: MongoDBMetadataUI = {
+      requires: {
+        mongodb: '>=4.2.0',
+        clientSideEncryption: true
+      }
+    };
+
+    describe('KMS Request', () => {
+      describe('Node.js resource: TLS file read', () => {
+        describe('when KMSRequest reads an infinite TLS file', () => {
+          it('the file read is interrupted by client.close()', async () => {
+            await runScriptAndGetProcessInfo(
+              'tls-file-read',
+              config,
+              async function run({ MongoClient, uri, log, chai, ClientEncryption, BSON }) {
+                const infiniteFile = '/dev/zero';
+
+                const kmsProviders = BSON.EJSON.parse(process.env.CSFLE_KMS_PROVIDERS);
+                const masterKey = {
+                  region: 'us-east-1',
+                  key: 'arn:aws:kms:us-east-1:579766882180:key/89fcc2c4-08b0-4bd9-9f25-e30687b580d0'
+                };
+                const provider = 'aws';
+
+                const keyVaultClient = new MongoClient(uri);
+                await keyVaultClient.connect();
+                await keyVaultClient.db('keyvault').collection('datakeys');
+
+                const clientEncryption = new ClientEncryption(keyVaultClient, {
+                  keyVaultNamespace: 'keyvault.datakeys',
+                  kmsProviders
+                });
+                const dataKey = await clientEncryption.createDataKey(provider, { masterKey });
```

Comment:
Can we do all the data key setup in a `beforeHook`?  That avoids cluttering the test with setup logic

Reply:
If we used a `beforeHook`, then we'd have to pass it into the process somehow. Do you have any suggestions? Although it is cluttered, I think this is the most accurate way to go about this. I can also make a helper func 'encryptionSetup' within this function to make the code a bit clearer.